### PR TITLE
Add boolean fields and use them in taxonomy forms

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/1001

### Code Changes

* [x] Add a new component `CustomRadioGroup` for radio buttons that work with formik and all that
* [x] Use the new component to render the `legitimate_interest` and `automatic_decisions_or_profiling` fields
* [x] Changed the `extraFormFields` prop to `renderExtraFormFields` so that the form values can be passed in the case we want to conditionally render form fields. This was useful for `legitimate_interest` since only if that field is `true` do you need to fill in `legitimate_interest_impact_assessment`
* [x] Update tests

### Steps to Confirm

* [ ] New fields should show up when editing and adding taxonomy entities

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

https://user-images.githubusercontent.com/24641006/187709355-6bd6307b-e6df-452f-bb17-c26dc69ca8fb.mov

### CODE REVIEW

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx [hunk 1]
```diff
@@ -160,15 +169,22 @@ export const useDataUse = (): TaxonomyHookData<DataUse> => {
...
-      <CustomTextInput
-        name="legitimate_interest_impact_assessment"
-        label={labels.legitimate_interest_impact_assessment}
+      <CustomRadioGroup
+        name="legitimate_interest"
+        label={labels.legitimate_interest}
+        options={YesNoOptions}
       />
+      {formValues.legitimate_interest &&
+      formValues.legitimate_interest.toString() === "true" ? (
```
Comment (LKCSmith) [inline clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:182]: Nit, but does this work the same if you made it one line like `formValues.legitimate_interest?.toString() === "true" ? (`

File: clients/ctl/admin-ui/src/features/common/form/inputs.tsx [hunk 2]
```diff
@@ -411,3 +414,54 @@ export const CustomTextArea = ({
...
+  label: string;
+  options: Option[];
+}
+export const CustomRadioGroup = ({
+  label,
+  options,
+  ...props
+}: CustomRadioGroupProps & FieldHookConfig<string>) => {
+  const [field, meta] = useField(props);
+  const isInvalid = !!(meta.touched && meta.error);
```
Comment (LKCSmith) [inline clients/ctl/admin-ui/src/features/common/form/inputs.tsx:428]: Same here for the double `!!` -- or are we using the double negative for type conversions?

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx [hunk 3]
```diff
@@ -137,7 +143,10 @@ export const useDataUse = (): TaxonomyHookData<DataUse> => {
       legal_basis: du.legal_basis,
       special_category: du.special_category,
       recipients: du.recipients ?? [],
-      legitimate_interest: du.legitimate_interest,
+      legitimate_interest:
+        du.legitimate_interest == null
```
Comment (LKCSmith) [inline clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:147]: Should we use triple `===`? 👀

**Review by LKCSmith — APPROVED** [review]
Approved, some questions but nothing that blocks - more just curiosity!

File: clients/ctl/admin-ui/src/features/common/form/inputs.tsx (hunk 2 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/common/form/inputs.tsx:428]: yup yup exactly, for the type conversion to boolean to keep typescript happy

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx [hunk 4]
```diff
@@ -121,6 +123,10 @@ export const useDataUse = (): TaxonomyHookData<DataUse> => {
       formValues.legitimate_interest_impact_assessment === ""
         ? undefined
         : formValues.legitimate_interest_impact_assessment,
+    legitimate_interest: !!(
```
[the remark this answers is not in the document]
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:126]: this needs to be a boolean, so same thing as the other comment

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx (hunk 3 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:147]: yah I think there are different opinions on this. I've always gone with 'always use `===`, except `==` is useful if you want to see if a value is `undefined` **or** `null`'. more info on that in [this article](https://blog.webdevsimplified.com/2020-08/==-vs-===/) (see the "When to use Double Equals" section. the article is very strongly titled considering they don't actually recommend to _totally_ stop using double equals 😅 )

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx (hunk 1 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:182]: yeah I think you're right!

File: clients/ctl/admin-ui/src/features/common/form/inputs.tsx (hunk 2 above)
Comment (LKCSmith) [reply clients/ctl/admin-ui/src/features/common/form/inputs.tsx:428]: Gotcha! Thanks for confirming 😄

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx (hunk 3 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:147]: in this case it's only comparing to `undefined` though, so I'll change to `=== undefined` 👍

Comment (allisonking) [reply clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:146]: oh never mind... it turns out it _can_ be null or undefined, so the `== null` is still more handy here. thanks to the tests which caught that 😅

**Comment by allisonking** [comment]
Confirmed the two failing tests pass locally. As discussed in sprint planning today, will go ahead and merge this one while we awaiting a fix for what's going on with those failing tests. (related: https://github.com/ethyca/fides/pull/1026#issuecomment-1232431906)

**Pull request merged**